src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/IgnitionChamber.java

```diff
@@ -82,31 +85,43 @@ public static boolean useFlintAndSteel(Player p, Block smelteryBlock) {
         }
 
         // Check if the chamber contains a Flint and Steel
-        if (inv.contains(Material.FLINT_AND_STEEL)) {
-            ItemStack item = inv.getItem(inv.first(Material.FLINT_AND_STEEL));
-            ItemMeta meta = item.getItemMeta();
-
-            // Only damage the Flint and Steel if it isn't unbreakable.
-            if (!meta.isUnbreakable()) {
-                // Update the damage value
-                ((Damageable) meta).setDamage(((Damageable) meta).getDamage() + 1);
-
-                if (((Damageable) meta).getDamage() >= item.getType().getMaxDurability()) {
-                    // The Flint and Steel broke
-                    item.setAmount(0);
-                    SoundEffect.IGNITION_CHAMBER_USE_FLINT_AND_STEEL_SOUND.playAt(smelteryBlock);
-                } else {
-                    item.setItemMeta(meta);
-                }
-            }
-
-            SoundEffect.IGNITION_CHAMBER_USE_FLINT_AND_STEEL_SOUND.playAt(smelteryBlock);
-            return true;
-        } else {
+        if (!inv.contains(Material.FLINT_AND_STEEL)) {
             // Notify the Player there is a chamber but without any Flint and Steel
             Slimefun.getLocalization().sendMessage(p, "machines.ignition-chamber-no-flint", true);
             return false;
         }
+
+        ItemStack item = inv.getItem(inv.first(Material.FLINT_AND_STEEL));
+
+        // Only damage the Flint and Steel if it isn't unbreakable.
+        damageFlintAndSteel(item, smelteryBlock);
+
+        SoundEffect.IGNITION_CHAMBER_USE_FLINT_AND_STEEL_SOUND.playAt(smelteryBlock);
+        return true;
+    }
+
+    private static void damageFlintAndSteel(ItemStack flintAndSteel, Block smelteryBlock) {
+        ItemMeta meta = flintAndSteel.getItemMeta();
+        Damageable damageable = (Damageable) meta;
+
+        if (meta.isUnbreakable()) {
+            return;
+        }
+
+        Enchantment unbreaking = VersionedEnchantment.UNBREAKING;
+        int lvl = flintAndSteel.getEnchantmentLevel(unbreaking);
+
+        if (UnbreakingAlgorithm.TOOLS.evaluate(lvl)) {
+            damageable.setDamage(damageable.getDamage() + 1);
+        }
+
+        if (damageable.getDamage() >= flintAndSteel.getType().getMaxDurability()) {
```

Comment:
not sure how we should handle components here, since items can now have their own max durability, we might need to have a util somewhere that returns getType#getMaxDura if version is below components otherwise returns damageable.getMaxDamage() (or whatever the method is can't remember)

Reply:
A concrete example would be helpful, also i think it is too soon to consider this problem as i doubt that feature is fully implemented

Reply:
That feature is fully implemented and I have used it myself lol.

For example an addon adds "Flint and Damascus-Steel", it's normal flint and steel except has more durability using the new component system. Lets say it has 256 durability unlike the normal (64? I think).

If they put it in the ignition chamber once the durability is 192/256 or less it'll get removed even though it has more to spend

Reply:
ok didn't know that, when was that added?

Reply:
1.20.5+ when item components were introduced, any item regardless of type can be given durability